momento-functions/examples/turbopuffer-index-articles.rs

```diff
@@ -0,0 +1,283 @@
+            turbopuffer_endpoint.clone(),
+            [
+                ("Authorization".to_string(), turbopuffer_api_key.clone()),
+                ("Content-Type".to_string(), "application/json".to_string()),
+                ("Accept".to_string(), "*/*".to_string()),
+                (
+                    "User-Agent".to_string(),
+                    "momento-turbobuffer-example".to_string(),
+                ),
+            ],
+            json!({
+                "upsert_rows": turbopuffer_inputs,
+                "distance_metric": "cosine_distance",
+            }),
+        );
+        match result {
+            Ok(response) => {
+                log::debug!(
+                    "turbopuffer response: {} - {:?}",
+                    response.status,
+                    response.headers
+                );
+                if response.status != 200 {
+                    let message = format!(
+                        "Failed to index documents: {}",
+                        String::from_utf8(response.body).unwrap_or_default(),
+                    );
+                    return Ok(WebResponse::new().with_status(response.status).with_body(
+                        json!({
+                            "message": message,
+                        }),
+                    )?);
+                }
+            }
+            Err(e) => {
+                log::error!("Failed to index documents: {e:?}");
+                return Ok(WebResponse::new().with_status(500).with_body(json!({
+                    "message": e.to_string(),
+                }))?);
+            }
+        }
+    }
+    Ok(WebResponse::new().with_status(200).with_body(json!({
+        "message": "Documents indexed successfully",
+    }))?)
+}
+
+// ------------------------------------------------------
+// | Utility functions for convenience
+// ------------------------------------------------------
+
+fn get_embeddings(mut documents: Vec<String>, openai_key: String) -> WebResult<Vec<EmbeddingData>> {
+    log::debug!("getting embeddings for input");
+    for document in &mut documents {
+        if document.contains("\n") {
+            // openai guide currently says to replace newlines with spaces. This, then, must be how you get the cargo to come.
+            // https://platform.openai.com/docs/guides/embeddings
+            *document = document.replace("\n", " ");
+        }
+        if document.is_empty() {
+            // openai will fail to generate an embedding if no content is provided
+            *document = "no_content".to_string();
+        }
+    }
+
+    let result = momento_functions_host::http::post(
+        "https://api.openai.com/v1/embeddings",
+        [
+            ("authorization".to_string(), format!("Bearer {openai_key}")),
+            ("content-type".to_string(), "application/json".to_string()),
+        ],
+        // 1536 float32 for text-embedding-3-small
+        serde_json::json!({
+            "model": "text-embedding-3-small",
```

Comment:
For clarity let's extract the url and model name to static strings.

Reply:
https://github.com/momentohq/functions/pull/4/commits/6fff10202caceff0f804eed460052e2b4e291610